drivers/counter/counter_native_posix.c

```diff
@@ -105,14 +145,22 @@ static int ctr_set_alarm(const struct device *dev, uint8_t chan_id,
 		return -ENOTSUP;
 	}
 
-	pending_alarm = *alarm_cfg;
-	is_alarm_pending = true;
+	uint32_t ticks = alarm_cfg->ticks;
 
 	if (!(alarm_cfg->flags & COUNTER_ALARM_CFG_ABSOLUTE)) {
-		pending_alarm.ticks =
-			hw_counter_get_value() + pending_alarm.ticks;
+		ticks += hw_counter_get_value();
 	}
 
+	if (is_top_set && (ticks > top.ticks)) {
+		posix_print_warning("Alarm ticks %u exceed top ticks %u\n", ticks,
+				    top.ticks);
+		return -EINVAL;
+	}
+
+	pending_alarm = *alarm_cfg;
+	pending_alarm.ticks = ticks;
+	is_alarm_pending = true;
+
 	hw_counter_set_target(pending_alarm.ticks);
```

Comment:
This would seem incorrect, as alarm.ticks may be after a wrap (due to either top.ticks or TOP_VALUE) (see the expression in the other comment about finding the next hit)